### PR TITLE
Generate new deployment keys

### PR DESCRIPTION
**Pull request by thk123**
The script has suffered a little rot (no longer extracting the correct key id). I believe I've fixed this and made it more robust in the process. I also believe I have correctly updated the keys, though I'm a little concerned the size of the .xml.enc.has gone down by 32bytes and the encrypted key has halved.

### CODE REVIEW

**Comment by thk123** [comment]
When @sonodtt I will update the wiki with a couple of extra points: 
 - what a public key and a private key should look like (in case the script breaks again)
 - the fact that this process requires the password
 - exactly how to use the encrypt/decrypt scripts 
 - that you need to say y to deleting the keys
 - that the new key expires in Jan 2020

File: release_util_script_create_gpg.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ gpg --batch --gen-key gen-key-script
 #gpg --quick-gen-key 'Sonny Martin <sonny.martin@diffblue.com>'
 #--------------------------------------------------------------------------------
 ## get key id
-GPG_KEYID=$(gpg -K | head -n4 | tail -n1 | tr -d ' ')
+GPG_KEYID=$( gpg --list-keys "Sonny Martin" | head -n1 | grep -Po "(?<=4096R/)[A-Z0-9]+")
```
Comment (antlechner) [inline release_util_script_create_gpg.sh]: :grey_question: What is the difference between `-K` and `--list-keys`? The man page for gpg seems to suggest that `-K` lists all secret keys and `--list-keys` lists all public keys. I'm guessing the ID would be the same in both cases so it doesn't matter?
:question: I think I remember there being a problem with `grep -P` in models-library scripts on Macs, as Macs use a default version of grep that has strictly fewer features than GNU grep (but it's possible to install GNU grep using Homebrew). Should we test this script on a Mac, or just add a comment mentioning that there might be problems with this?

**Comment by thk123** [comment]
So a few more things I had to do:
 - replace the gpg with the new gpg key
 - say yes to deleting the keys when running the script.

**Comment by thk123** [comment]
Replaced by #65 - sorry could I get you both to review that instead (almost the same, except I think I've got the keys right this time!

File: release_util_script_create_gpg.sh (hunk 1 above)
Comment (thk123) [reply release_util_script_create_gpg.sh]: >  I'm guessing the ID would be the same in both cases so it doesn't matter

Thanks for googling - so yes in this case public keys are sufficent, I find single letter flags a bit awkward to google so prefer to leave as I have it here. 

> `grep -P` on mac

Good question @sonodtt would you be able to check whether my replacement works on Mac?

Comment (antlechner) [reply release_util_script_create_gpg.sh]: @sonodtt (also checking which version of grep you are using - it should be the default one on OS X, I forgot what it's called but I remember it not having "GNU" anywhere in its version name)

**Pull request closed without merging**